src/inputs/AddressInput/index.tsx

```diff
@@ -0,0 +1,222 @@
+  helperText?: string | undefined;
+};
+
+function AddressInput({
+  id,
+  name,
+  address,
+  networkPrefix,
+  showNetworkPrefix,
+  label,
+  placeholder,
+  defaultValue,
+  disabled,
+  onChangeAddress,
+  getAddressFromDomain,
+  inputAdornment,
+  showLoadingSpinner,
+  error = '',
+  helperText,
+  ...rest
+}: AddressInputProps): React.ReactElement {
+  const [prefix, setPrefix] = useState(showNetworkPrefix ? networkPrefix : '');
+  const [isLoadingENSResolution, setIsLoadingENSResolution] = useState(false);
+
+  function onChange(e: ChangeEvent<HTMLInputElement>) {
+    const value = trimSpaces(e.target.value);
+
+    const prefix = getNetworkPrefix(value);
+    const address = getAddressWithoutNetworkPrefix(value);
+
+    const isValidPrefix =
+      value.includes(':') && !value.startsWith(':') && prefix === networkPrefix;
+
+    if (isValidPrefix) {
+      setPrefix(prefix);
+      onChangeAddress(address);
+    } else {
+      setPrefix('');
+      onChangeAddress(value);
+    }
+  }
+
+  // checksum valid addresses
+  useEffect(() => {
+    if (isValidAddress(address) && !isChecksumAddress(address)) {
+      onChangeAddress(checksumAddress(address));
+    }
+  }, [address, onChangeAddress]);
+
+  // ENS resolution
+  useEffect(() => {
+    async function resolveDomainName(ENSName: string) {
+      try {
+        setIsLoadingENSResolution(true);
+        const address = (await getAddressFromDomain?.(ENSName)) as string;
+        onChangeAddress(address);
+        setPrefix(showNetworkPrefix ? networkPrefix : '');
+      } catch (e) {
+        onChangeAddress(ENSName);
+        setPrefix('');
+      } finally {
+        setIsLoadingENSResolution(false);
+      }
+    }
+
+    const isEnsName = isValidEnsName(address);
+
+    if (isEnsName && getAddressFromDomain) {
+      throttle(() => resolveDomainName(address));
+    }
+  }, [
+    address,
+    getAddressFromDomain,
+    networkPrefix,
+    showNetworkPrefix,
+    onChangeAddress,
+  ]);
+
+  const adornment =
+    isLoadingENSResolution || showLoadingSpinner ? (
+      <InputAdornment position="end">
+        <CircularProgress size="16px" />
+      </InputAdornment>
+    ) : (
+      (inputAdornment as ReactElement)
+    );
+
+  const hasError = !!error;
+
+  const value = addNetworkPrefix(address, prefix);
+
+  return (
+    <TextField
+      id={id || name}
+      value={value}
+      hiddenLabel={!value}
+      label={label}
+      helperText={hasError ? error : helperText}
+      placeholder={placeholder}
+      name={name}
+      defaultValue={defaultValue}
+      disabled={disabled || isLoadingENSResolution}
+      onChange={onChange}
+      InputProps={{ endAdornment: adornment }}
+      error={hasError}
+      color="primary"
+      variant="filled"
+      spellCheck={false}
+      InputLabelProps={{
+        shrink: true,
+      }}
+      {...rest}
+    />
+  );
+}
+
+export default AddressInput;
+
+const TextField = styled(TextFieldMui)`
```

Comment:
I would define this on top or into a separate file. It's a long component and wasn't expecting to find it here

Reply:
Also, was there some issue why we didn't use the existing TextField in `src/input/TextField/index`

Reply:
Added a new `TextFieldInput` and deprecated the old `TextField` component because is coupled to Final Form